src/Giraffe/ResponseWriters.fs

```diff
@@ -157,9 +160,34 @@ type HttpContext with
         task {
             this.SetContentType "application/json; charset=utf-8"
             this.SetHttpHeader "Transfer-Encoding" "chunked"
+            let writeChunk (strOut : string) = task {
+                // Write the length in hex
+                do! this.Response.WriteAsync (sprintf "%s\r\n" (strOut.Length.ToString("X")))
+                // Write the chunk
+                do! this.Response.WriteAsync (sprintf "%s\r\n" strOut)
+            }
+            let writeEndChunk () = task {
+                do! this.Response.WriteAsync ("0\r\n\r\n")
+            }
             if this.Request.Method <> HttpMethods.Head then
-                let serializer = this.GetJsonSerializer()
-                do! serializer.SerializeToStreamAsync dataObj this.Response.Body
+                let pool = ArrayPool<char>.Shared
+                let buffer = pool.Rent 4096
+                try
+                    use memoryStream = recyclableMemoryStreamManager.GetStream()
+                    let serializer = this.GetJsonSerializer()
+                    do! serializer.SerializeToStreamAsync dataObj memoryStream
```

Comment:
Unfortunately this design does suffer from copying to two separate `RecyclableMemoryStreams`